lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -29,4 +37,51 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @GetMapping("/items/{id}")
+  public ResponseEntity<MediaItemResponse> getItemById(@PathVariable("id") UUID id) {
+    String stringId = id.toString();
+    SearchCriteria searchCriteria = SearchCriteria.builder().id(stringId).build();
+
+    Set<MediaItem> foundItems = library.search(searchCriteria);
+
+    if (foundItems.isEmpty()) {
+      return ResponseEntity.notFound().build();
+    }
+
+    MediaItem item = foundItems.iterator().next();
+    MediaItemResponse response = MediaItemResponse.from(item);
+
+    return ResponseEntity.ok(response);
+  }
+
+  @PostMapping("/items")
+  public CreateMediaItemResponse createMediaItem(@RequestBody @Valid CreateMediaItemRequest request)
+      throws MethodArgumentNotValidException {
+
+    MediaItemRequest itemRequest = request.getItem();
+
+    System.out.println(request);
+
+    var item = MediaItemRequest.asMediaItem(itemRequest);
+
+    library.addMediaItem(item, librarian);
+
+    var response =
+        CreateMediaItemResponse.builder().item(getItemById(item.getId()).getBody()).build();
+
+    return response;
+  }
+
+  @DeleteMapping("/items/{id}")
+  public ResponseEntity<Void> deleteItem(@PathVariable("id") UUID id) {
+
+    if (getItemById(id).getBody() == null) {
```

Comment:
Naw, this is weird. Use the library to check if the item exists (you can use `library.search`).

Reply:
Changed